src/aggregator/client/queue.go

```diff
@@ -132,15 +135,20 @@ func newInstanceQueue(instance placement.Instance, opts Options) instanceQueue {
 		conn               = newConnection(instance.Endpoint(), connOpts)
 		iOpts              = opts.InstrumentOptions()
 		queueSize          = opts.InstanceQueueSize()
+		maxBatchSize       = opts.MaxBatchSize()
+		writeInterval      = opts.BatchFlushDeadline()
 	)
 	q := &queue{
-		dropType: opts.QueueDropType(),
-		log:      iOpts.Logger(),
-		metrics:  newQueueMetrics(iOpts.MetricsScope(), queueSize),
-		instance: instance,
-		conn:     conn,
-		bufCh:    make(chan protobuf.Buffer, queueSize),
-		doneCh:   make(chan struct{}),
+		dropType:           opts.QueueDropType(),
+		log:                iOpts.Logger(),
+		metrics:            newQueueMetrics(iOpts.MetricsScope(), queueSize),
+		instance:           instance,
+		conn:               conn,
+		bufCh:              make(chan protobuf.Buffer, queueSize),
+		doneCh:             make(chan struct{}),
+		maxBatchSize:       maxBatchSize,
+		batchFlushDeadline: writeInterval,
+		buf:                make([]byte, 0, maxBatchSize),
```

Comment:
Is this one queue per backend * host?

Reply:
one per shard x instance, from the looks of it